src/server/protocol.d.ts

```diff
@@ -0,0 +1,820 @@
+/** 
+  * Declaration module describing the TypeScript Server protocol 
+  */
+declare module ts.server.protocol {
+    /** 
+      * A TypeScript Server message 
+      */
+    export interface Message {
+        /** 
+          * Sequence number of the message 
+          */
+        seq: number;
+
+        /**
+          * One of "request", "response", or "event" 
+          */
+        type: string;

```

Comment:
What is 'event' used for?  Why does hte message need this type?  Isn't it clear based on what direction we're sending things?